File: creational/prototype/prototype.dart
class Student {
    String name;
    int age;

    Student(this.name, this.age);

    Student.fromSource(Student student) {
        name = student.name;
        age = student.age;
    }

    @Override
    Student clone() {
        return Student.fromSource(this);
    }
}

void main() {
    var student1 = Student('PNG', 10);
    var student2 = student1.clone();
    print(student1);
    print(student2);
}

File: creational/prototype/prototype.py
import copy


class Student(object):

    def __init__(self, name: str = 'PNG', age: int = 10):
        self.name = name
        self.age = age

    def clone(self):
        return copy.deepcopy(self)


if __name__ == '__main__':
    student1 = Student('Phong', 23)
    print(student1.name)
    student2 = student1.clone()
    print(student2.name)
